### DISCUSSION TITLE
!tapi-tbd error compiling on macOS Catalina, version 10.15.7

### DISCUSSION THREAD

**Discussion by mangerij** [idaholab/moose#16353] [Q&A Getting Started] [answered]
Hello,

A friend of mine is compiling MOOSE on macOS Catalina, version 10.15.7. He uses the conda approach to install and gets the following compile error:

```
Linking Library /Users/mg/programs/moose/framework/contrib/pcre/libpcre-opt.la...
Linking Library /Users/mg/programs/moose/framework/contrib/gtest/libgtest.la...
Linking Library /Users/mg/programs/moose/framework/contrib/hit/libhit-opt.la...
Building and linking /Users/mg/programs/moose/framework/../python/pyhit/hit.so...
ld: unsupported tapi file type '!tapi-tbd' in YAML file '/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib/libm.tbd' for architecture x86_64
clang-9: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [/Users/mg/programs/moose/framework/contrib/gtest/libgtest.la] Error 1
make: *** Waiting for unfinished jobs....
ld: unsupported tapi file type '!tapi-tbd' in YAML file '/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib/libm.tbd' for architecture x86_64
clang-9: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [/Users/mg/programs/moose/framework/contrib/hit/libhit-opt.la] Error 1
ld: unsupported tapi file type '!tapi-tbd' in YAML file '/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib/libm.tbd' for architecture x86_64
clang-9: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [/Users/mg/programs/moose/framework/contrib/pcre/libpcre-opt.la] Error 1
ld: unsupported tapi file type '!tapi-tbd' in YAML file '/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib/libSystem.tbd' for architecture x86_64
clang-9: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [/Users/mg/programs/moose/framework/../python/pyhit/hit.so] Error 1
```

I googled this error and found some suggestions are to update the Xcode library on the Mac. He did so and still gets the problem. I'm not very familiar with compiling on Mac so are there any suggestions here? I'll direct him to this thread if there are any more questions to answer about his machine/build.

Thanks

**Comment by cticenhour** [comment, marked as answer]
I am working on updating the MOOSE conda packages for compatibility with Xcode / Command Line Tools version 12 that will fix this issue in #16295. In the meantime, your friend needs to downgrade his Command Line Tools to version 11.5. He can do this by performing the following:

- Remove the Command Line Tools directory to remove the newer version: `sudo rm -rf /Library/Developer/CommandLineTools`
- Go to https://www.apple.com/developer/downloads, sign in with your Apple ID, and search for "Command Line Tools 11.5"
- Download the dmg file and install by double-clicking on it and following the prompts.
- Restart your terminal! 
- You should then clean your MOOSE directory. Make sure you have stashed or committed any work you want to save. Then, perform the command `git clean -xfd` in `~/programs/moose/`
- Try building MOOSE again.

**Reply by jerenmb** [reply]
I had the same issue and this solved the problem. Thanks!

**Reply by cticenhour** [reply]
@mangerij @jerenmb The updates to MOOSE and our conda package that fix this issue have been merged into MOOSE master. You should feel free to update your Xcode and/or Command Line Tools after performing these updates. See the announcement here: https://groups.google.com/g/moose-announce/c/t4ogAXgU26Y

If you have any further issues, please feel free to reach out.
